frontend/src/hooks/useAgents.tsx

```
@@ -0,0 +1,135 @@
+
+type AgentAction = 
+  | { type: 'ADD_AGENT'; payload: Agent }
+  | { type: 'UPDATE_AGENT'; payload: { id: string; updates: Partial<Agent> } }
+  | { type: 'ADD_MERCHANT'; payload: { agentId: string; merchantAddress: string } }
+  | { type: 'REMOVE_MERCHANT'; payload: { agentId: string; merchantAddress: string } }
+  | { type: 'LOAD_AGENTS'; payload: Agent[] }
+
+function agentReducer(state: Agent[], action: AgentAction): Agent[] {
+  switch (action.type) {
+    case 'ADD_AGENT':
+      const newAgents = [...state, action.payload]
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)